src/frontend/js/data/SessionProvider/index.spec.tsx

```diff
@@ -0,0 +1,193 @@
+import { QueryClient, QueryClientProvider } from 'react-query';
+import faker from 'faker';
+import fetchMock from 'fetch-mock';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import {
+  ContextFactory as mockContextFactory,
+  PersistedClientFactory,
+  QueryStateFactory,
+} from 'utils/test/factories';
+import createQueryClient from 'utils/react-query/createQueryClient';
+import { Deferred } from 'utils/test/deferred';
+import { REACT_QUERY_SETTINGS } from 'settings';
+import BaseSessionProvider from './BaseSessionProvider';
+import { useSession } from '.';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockContextFactory({
+    authentication: {
+      endpoint: 'https://endpoint.test',
+      backend: 'openedx-hawthorn',
+    },
+    joanie_backend: {
+      endpoint: 'https://joanie.test',
+    },
+  }).generate(),
+}));
+
+describe('SessionProvider', () => {
+  const queryClient = createQueryClient({ persistor: true });
+
+  const wrapper = ({
+    client = queryClient,
+    children,
+  }: React.PropsWithChildren<{ client: QueryClient }>) => (
+    <QueryClientProvider client={client}>
+      <BaseSessionProvider>{children}</BaseSessionProvider>
+    </QueryClientProvider>
+  );
+
+  beforeEach(() => {
+    jest.useFakeTimers('modern');
+    jest.resetModules();
+    fetchMock.restore();
+    queryClient.clear();
+    sessionStorage.clear();
+  });
+
+  afterEach(() => {
+    jest.clearAllTimers();
+  });
+
+  it('uses BaseSessionProvider if joanie is disabled', async () => {
+    jest.doMock('../../utils/api/joanie.ts', () => ({
+      isJoanieEnabled: false,
+    }));
+    jest.doMock('./BaseSessionProvider', () => ({
+      __esModule: true,
+      default: () => 'BaseSessionProvider',
+    }));
+    jest.doMock('./JoanieSessionProvider', () => ({
+      __esModule: true,
+      default: () => 'JoanieSessionProvider',
+    }));
+    const { SessionProvider: Provider } = await import('.');
```

Comment:
As this is node code, you can also use `require` instead of `await import` here.